storage/postgres/log_storage.go

```diff
@@ -0,0 +1,1011 @@
+	var logRoot types.LogRootV1
+	if err := logRoot.UnmarshalBinary(root.LogRoot); err != nil {
+		glog.Warningf("Failed to parse log root: %x %v", root.LogRoot, err)
+		return err
+	}
+	if len(logRoot.Metadata) != 0 {
+		return fmt.Errorf("unimplemented: postgres storage does not support log root metadata")
+
+	}
+	//get a json copy of the tree_head
+	data, _ := json.Marshal(logRoot)
+	t.tx.ExecContext(
+		ctx,
+		"update trees set current_tree_data = $1,root_signature = $2 where tree_id = $3",
+		data,
+		root.LogRootSignature,
+		t.treeID)
+	res, err := t.tx.ExecContext(
+		ctx,
+		insertTreeHeadSQL,
+		t.treeID,
+		logRoot.TimestampNanos,
+		logRoot.TreeSize,
+		logRoot.RootHash,
+		logRoot.Revision,
+		root.LogRootSignature)
+	if err != nil {
+		glog.Warningf("Failed to store signed root: %s", err)
+	}
+
+	return checkResultOkAndRowCountIs(res, err, 1)
+}
+
+func (t *logTreeTX) getLeavesByHashInternal(ctx context.Context, leafHashes [][]byte, tmpl *sql.Stmt, desc string) ([]*trillian.LogLeaf, error) {
+	stx := t.tx.StmtContext(ctx, tmpl)
+	defer stx.Close()
+
+	var args []interface{}
+	for _, hash := range leafHashes {
+		args = append(args, interface{}([]byte(hash)))
+	}
+	args = append(args, interface{}(t.treeID))
+	rows, err := stx.QueryContext(ctx, args...)
+	if err != nil {
+		glog.Warningf("Query() %s hash = %v", desc, err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	// The tree could include duplicates so we don't know how many results will be returned
+	var ret []*trillian.LogLeaf
+	for rows.Next() {
+		leaf := &trillian.LogLeaf{}
+		// We might be using a LEFT JOIN in our statement, so leaves which are
+		// queued but not yet integrated will have a NULL IntegrateTimestamp
+		// when there's no corresponding entry in SequencedLeafData, even though
+		// the table definition forbids that, so we use a nullable type here and
+		// check its validity below.
+		var integrateTS sql.NullInt64
+		var queueTS int64
+
+		if err := rows.Scan(&leaf.MerkleLeafHash, &leaf.LeafIdentityHash, &leaf.LeafValue, &leaf.LeafIndex, &leaf.ExtraData, &queueTS, &integrateTS); err != nil {
+			glog.Warningf("LogID: %d Scan() %s = %s", t.treeID, desc, err)
+			return nil, err
+		}
+		var err error
+		leaf.QueueTimestamp, err = ptypes.TimestampProto(time.Unix(0, queueTS))
+		if err != nil {
+			return nil, fmt.Errorf("got invalid queue timestamp: %v", err)
+		}
+		if integrateTS.Valid {
+			leaf.IntegrateTimestamp, err = ptypes.TimestampProto(time.Unix(0, integrateTS.Int64))
+			if err != nil {
+				return nil, fmt.Errorf("got invalid integrate timestamp: %v", err)
+			}
+		}
+
+		if got, want := len(leaf.MerkleLeafHash), t.hashSizeBytes; got != want {
+			return nil, fmt.Errorf("LogID: %d Scanned leaf %s does not have hash length %d, got %d", t.treeID, desc, want, got)
+		}
+
+		ret = append(ret, leaf)
+	}
+
+	return ret, nil
+}
+
+func (t *readOnlyLogTX) GetUnsequencedCounts(ctx context.Context) (storage.CountByLogID, error) {
```

Comment:
This method has been removed from the storage interface so can be removed here too.